lib/libzutil/Makefile.am

```diff
@@ -45,7 +45,8 @@ libzutil_la_LIBADD = \
 
 if BUILD_LINUX
 libzutil_la_LIBADD += \
-        $(abs_top_builddir)/lib/libefi/libefi.la
+        $(abs_top_builddir)/lib/libefi/libefi.la \
+	-lrt
```

Comment:
On FreeBSD and I think OSX too, lio_listio will be more efficient than using a thread pool.  On Linux it's probably about the same.

Reply:
One other possible concern with using AIO here is that there may be some niche filesystem, or maybe even block device drivers, on Linux which don't fully support it.  There are so many it's impossible to keep track.  It's _possible_ this could break some existing setups.

Reply:
Well, I know of one.  On FreeBSD, using file-backed vdevs with the files stored on NFS will fail.  There's a sysctl that you can change to allow it, though.  But nobody has complained about it since that change originally landed in FreeBSD in 2017.

I could easily add a serialized fallback path for the case of ENOSYS or EAGAIN.  But my concern is that using the fallback path for all error types would double the amount of disk activity even for non-recoverable errors like EACCES.  On Linux, I don't think there should be trouble with any particular file system, because POSIX AIO is implemented by glibc using threads.  So as long as librt is available, it should work.

Reply:
Good point.  Then I'm happy with this as along as we confirm we're not causing a real problem for macos.

Reply:
I didn't even realize that OSX was supported.  It isn't listed at https://openzfs.github.io/openzfs-docs/Getting%20Started/index.html .  And I see that Buildbot isn't testing on OSX.  What does it take to test on that platform?  I don't have any macs myself.

Reply:
@asomers it's not yet!  But there is [work underway to add it](https://github.com/openzfs/zfs/pull/11273), and it'd be nice to avoid making more work for our future selves if we can.

Reply:
It would definitely be helpful if you could test this PR on OSX.  If you don't want to run the full test suite, then just do "zpool import" with both file-backed and disk-backed pools.  That should be sufficient.  And there's a Windows port too now?  Sheesh; I had no idea.

Reply:
I encountered no issues:
```
# ./scripts/cmd-macos.sh zpool import

   pool: BOOM
     id: 5628038252777279005
  state: ONLINE

	BOOM                                          ONLINE
	  media-3628A8FE-7BD0-8B42-89B9-6D46D57607EF  ONLINE

# ./scripts/cmd-macos.sh zpool import -d ~/
   pool: test
     id: 7967859940384804015
  state: ONLINE
 action: The pool can be imported using its name or numeric identifier.
 config:

	test                       ONLINE
	  /Users/lundman/disk.img  ONLINE
```

It is interesting to note that userland still uses `module/os/linux/zfs/vdev_file.c` on all platforms, not sure how I feel about that yet. But that isn't connected to the changes you made.

Reply:
> still uses module/os/linux/zfs/vdev_file.c on all platforms

Yeah, that's odd.  We should probably either make it clear it's common by moving it to `module/zfs/vdev_file.c`, or decide it really not and have per-platform versions.

Reply:
I came across it when working on Windows, where vdev_file.c (was) quite different (now in 2.0 port uses zfs_file_* API). I believe one other os/linux file is also taken from Linux, was it abd_os.c or arc_os.c hmm...   At the time I couldn't get my own vdev_file.c to work in userland (didn't want to spend more than 5 mins shaving yaks at that time). It is clearly working at the moment, but could be some Linux specific changes in future could surprise someone. So a userland only vdev_file.c could work, or, I'll try again to switch to os/macos/vdev_file (and windows+Freebsd).